src/vos/vos_dtx.c

```diff
@@ -1627,26 +1593,28 @@ vos_dtx_prepared(struct dtx_handle *dth)
 	    (dth->dth_modification_cnt > 0))
 		dth->dth_sync = 1;
 
-	DAE_OID_CNT(dae) = dth->dth_oid_cnt;
 	if (dth->dth_oid_array != NULL) {
-		D_ASSERT(dth->dth_oid_cnt != 0);
+		D_ASSERT(dth->dth_oid_cnt > 0);
 
+		dae->dae_oid_cnt = dth->dth_oid_cnt;
 		if (dth->dth_oid_cnt == 1) {
-			DAE_OID_INLINE(dae) = dth->dth_oid_array[0];
+			dae->dae_oid_inline = dth->dth_oid_array[0];
+			dae->dae_oids = &dae->dae_oid_inline;
 		} else {
 			size = sizeof(daos_unit_oid_t) * dth->dth_oid_cnt;
-
-			rec_off = umem_zalloc(umm, size);
-			if (umoff_is_null(rec_off)) {
-				D_ERROR("No space to store DTX OIDs "DF_DTI"\n",
-					DP_DTI(&DAE_XID(dae)));
-				return -DER_NOSPACE;
+			D_ALLOC_ARRAY(dae->dae_oids, size);
```

Comment:
above size = sizeof(daos_unit_oid_t) * dth->dth_oid_cnt; 
why still use D_ALLOC_ARRAY()?

Reply:
Typo, good catching it.